pkg/controller/kabaneroplatform/kappnav.go

```diff
@@ -19,137 +15,36 @@ var kanlog = rlog.Log.WithName("kabanero-kappnav")
-	// We need to create the structure here before we use it.
+	// KNavApp is optional.  We're basically just reporting if we found it.
 	status := kabanerov1alpha1.KappnavStatus{}
 	k.Status.Kappnav = &status
-	k.Status.Kappnav.Version = currentVersion
 	k.Status.Kappnav.ErrorMessage = ""
 	k.Status.Kappnav.Ready = "False"
 
-	clientset, err := getClient()
-	if err != nil {
-		k.Status.Landing.Ready = "False"
-		k.Status.Landing.ErrorMessage = "Failed to obtain client deployments interface."
-		return false, err
-	}
-
-	deploymentClient := clientset.AppsV1().Deployments(k.GetNamespace())
-	deployment, err := deploymentClient.Get("helm-operator", metav1.GetOptions{})
-
-	if err != nil {
-		if errors.IsNotFound(err) {
-			if !k.Spec.Kappnav.Enable {
-				k.Status.Kappnav = nil
-				return true, nil
-			}
-		}
-		message := "Unable to retrieve the kappnav helm-operator deployment object."
-		kanlog.Error(err, message)
-		k.Status.Kappnav.ErrorMessage = message + ": " + err.Error()
-		return false, err
-	}
-
-	conditions := deployment.Status.Conditions
-	// Find the condition type of 'Available' and it's value.
-	ready := false
-	for _, condition := range conditions {
-		if strings.ToLower(string(condition.Type)) == "available" {
-			if strings.ToLower(string(condition.Status)) == "true" {
-				ready = true
-				k.Status.Kappnav.Ready = "True"
-			} else {
-				k.Status.Kappnav.ErrorMessage = condition.Message
-			}
-
-			break
-		}
-	}
-
-	return ready, err
-}
-
-// Performs cleanup processing.
-func processCleanup(k *kabanerov1alpha1.Kabanero) error {
-	clientset, err := getClient()
-	deploymentClient := clientset.AppsV1().Deployments(k.GetNamespace())
-	if err != nil {
-		return err
-	}
-
-	deletePolicy := metav1.DeletePropagationForeground
-	err = deploymentClient.Delete("helm-operator", &metav1.DeleteOptions{
-		PropagationPolicy: &deletePolicy})
-	if err != nil {
-		if !errors.IsNotFound(err) {
-			return err
-		}
-	}
-
-	saClient := clientset.CoreV1().ServiceAccounts(k.GetNamespace())
-	err = saClient.Delete("helm-operator", &metav1.DeleteOptions{
-		PropagationPolicy: &deletePolicy})
-
-	if err != nil {
-		if errors.IsNotFound(err) {
-			return nil
-		}
-	}
-
-	return err
-}
-
-// Returns a Clientset object.
-func getClient() (*kubernetes.Clientset, error) {
-	// Create a clientset to drive API operations on resources.
-	config, err := clientcmd.BuildConfigFromFlags("", "")
-	if err != nil {
-		return nil, err
-	}
-
-	clientset, err := kubernetes.NewForConfig(config)
-	if err != nil {
-		return nil, err
-	}
-
-	return clientset, err
+	// We're looking in the default location for KAppNav.
+	u := &unstructured.Unstructured{}
+	u.SetGroupVersionKind(schema.GroupVersionKind{
+		Kind:    "KAppNav",
+		Group:   "charts.helm.k8s.io",
+		Version: "v1alpha1",
+	})
+	err := c.Get(context.Background(), client.ObjectKey{
+		Namespace: "kappnav",
+		Name:      "instance",
+	}, u)
+	
+	if err == nil {
```

Comment:
Just one comment. Checking that the instance exists to report that KAppNav is ready may not reflect the actual readiness of KAppNav.

Reply:
I will open up a follow-on issue to give a better status report.  I will look at your PR that we canceled, to see how you were reading the status etc from the KAppNav CR instance.